File: ulohy/27-prijimaci-rizeni/reseni/main.cpp
#include <iostream>
#include <string>
#include <vector>

int main() {
    int n;
    std::cin >> n;

    if (n < 1) {
        return 0;
    }

    std::vector<std::string> names(n);
    std::vector<double> scores(n);

    double sum = 0;

    for (int i = 0; i < n; i++) {
        std::cin >> names[i];
        std::cin >> scores[i];
        sum += scores[i];
    }

    double average = sum / n;

    for (int i = 0; i < n; i++) {
        if (scores[i] >= average) {
            std::cout << names[i] << std::endl;
        }
    }

    return 0;
}

File: ulohy/27-prijimaci-rizeni/reseni/main.cs
using System;

internal class Program
{
    static void Main(string[] args)
    {
        int n = int.Parse(Console.ReadLine());

        if (n < 1)
        {
            return;
        }

        string[] names = new string[n];
        double[] scores = new double[n];

        double sum = 0;

        for (int i = 0; i < n; i++)
        {
            Console.ReadLine();
            names[i] = Console.ReadLine();
            scores[i] = double.Parse(Console.ReadLine());
            sum += scores[i];
        }

        double average = sum / n;

        for (int i = 0; i < n; i++)
        {
            if (scores[i] >= average)
            {
                Console.WriteLine(names[i]);
            }
        }
    }
}
